internal/worker/api/server.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+
+	"github.com/renatoaguimaraes/job-scheduler/internal/worker/proto"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+func loadTLSCredentials(conf conf.Config) (credentials.TransportCredentials, error) {
+	// Load certificate of the CA who signed client's certificate
+	pemClientCA, err := ioutil.ReadFile(conf.ClientCA)
+	if err != nil {
+		return nil, err
+	}
+	// Certification pool to append client CA's certificate
+	certPool := x509.NewCertPool()
+	if !certPool.AppendCertsFromPEM(pemClientCA) {
+		return nil, fmt.Errorf("failed to add client CA's certificate")
+	}
+	// Load server's certificate and private key
+	serverCert, err := tls.LoadX509KeyPair(conf.ServerCertificate, conf.ServerKey)
+	if err != nil {
+		return nil, err
+	}
+	// Configure credentials to require and verify the client cert
+	config := &tls.Config{
+		Certificates: []tls.Certificate{serverCert},
+		ClientAuth:   tls.RequireAndVerifyClientCert,
+		ClientCAs:    certPool,
+	}
+	return credentials.NewTLS(config), nil
+}
+
+func StartGRPCServer(conf conf.Config) {
+	lis, err := net.Listen("tcp", conf.ServerAddress)
+	if err != nil {
+		log.Fatal("server can't start: ", err)
```

Comment:
Don't use log.Fatal in the library code, return an error and let the caller (server cmd package in this case) handle it.

Reply:
Ok, I will remove all occurrences and let's the caller to handle it.